dispatcher/brokers/pg_notify.py

```diff
@@ -129,6 +161,37 @@ def get_unlisten_query(self) -> psycopg.sql.SQL:
         """Stops listening on all channels for current session, see pg_notify docs"""
         return psycopg.sql.SQL("UNLISTEN *;")
 
+    async def initiate_self_check(self) -> None:
+        if self.self_check_status == BrokerSelfCheckStatus.IN_PROGRESS:
+            # another self-check message is in flight
+            raise RuntimeError(f'failed to initiate self check for broker {self.broker_id}')
+
+        try:
+            await self.apublish_message(self.self_check_channel, json.dumps({'self_check': True, 'sent': time.time(), 'task': f'lambda: "{self.broker_id}"'}))
+            self.self_check_status = BrokerSelfCheckStatus.IN_PROGRESS
+        except Exception as error:
+            raise RuntimeError(f'failed to send self check message: {error}')
```

Comment:
Mind if we get rid of this try-except block?

If this raises an exception, I would prefer to treat that as a failure of the check outright.

Reply:
Agree, this block actually doesn't do anything, it just catches and re-raises.